src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -1395,6 +1398,33 @@ public long getIdLong()
         return new AuditableRestActionImpl<>(getJDA(), route);
     }
 
+    @Nonnull
+    @Override
+    public AuditableRestAction<Void> timeoutUntilById(@Nonnull String userId, @Nonnull TemporalAccessor temporal)
+    {
+        Checks.isSnowflake(userId, "User ID");
+        Checks.notNull(temporal, "Temporal");
+        OffsetDateTime date = TimeUtil.toOffsetDateTime(temporal);
+        Checks.check(date.isAfter(OffsetDateTime.now()), "Cannot put a member in time out with date in the past. Provided: %s", date);
+        Checks.check(date.isBefore(OffsetDateTime.now().plusDays(28)), "Cannot put a member in time out for more than 28 days. Provided: %s", date);
+        checkPermission(Permission.MODERATE_MEMBERS);
+
+        DataObject body = DataObject.empty().put("communication_disabled_until", date.toString());
+        Route.CompiledRoute route = Route.Guilds.MODIFY_MEMBER.compile(getId(), userId);
+        return new AuditableRestActionImpl<>(getJDA(), route, body);
+    }
+
+    @Nonnull
+    @Override
+    public AuditableRestAction<Void> removeTimeoutById(@Nonnull String userId)
```

Comment:
I think this can be simplified by just allowing null for `timeoutUntilById` (similar to how we implemented `moveVoiceMember` and `kickVoiceMember`). Then this method can be default implemented as `timeoutUntilById(userId, null)`